homeassistant/components/rest/binary_sensor.py

```diff
@@ -88,7 +100,7 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     # No need to update the sensor now because it will determine its state
     # based in the rest resource that has just been retrieved.
-    add_entities([RestBinarySensor(hass, rest, name, device_class, value_template)])
+    add_entities([RestBinarySensor(hass, rest, name, device_class, value_template,)])
```

Comment:
You haven't copied all the parts of how Rest sensor does it. You now only render it once during setup and not on every update.

Reply:
Hello @balloob!

Thanks for Your intervention and time spent. I not quiet sure I understood.
I have modified update method to be more similar as rest_sensor. It will be great if You have time and review changes. If i am wrong in understanding PLEASE lhow me the right way.

About testing. Maybe there is some docs I can read about how to integrate it? I have a n IP camera with movement sensor. And I read this sensor by rest.

Reply:
You need to pass the resource_template to the class and need to render it. 

Tests are here https://github.com/home-assistant/core/blob/dev/tests/components/rest/test_binary_sensor.py

More info on testing in our dev docs at https://developer.home-assistant.io

Reply:
Hope I did everything like it shoud be...